File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
import java.util.*;

class Main {
    public static void main(String args[]) throws Exception {
        Scanner kb = new Scanner(System.in);
        int a = kb.nextInt();
        int b = kb.nextInt();
        int max=gcd(a,b); // 최대공약수
        int min=(a*b)/max;
        System.out.println(max);
        System.out.print(min);
    }

    static int gcd(int a, int b){
        if(b==0) return a;
        return gcd(b,a%b);
    }
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.swift
func solution(_ n:Int, _ m:Int) -> [Int] {
    var x = gcd(n, m)
    var y = min(n, m)
    return [x, y]
}

func gcd(_ a: Int,_ b: Int) -> Int {
    if b == 0 { return a }
    return gcd(b, a%b)
}

func min(_ a: Int,_ b: Int) -> Int {
    return a*b / gcd(a,b)
}